compiler/rustc_codegen_llvm/src/allocator.rs

```diff
@@ -114,8 +118,7 @@ pub(crate) unsafe fn codegen(
         attributes::emit_uwtable(llfn, true);
     }
 
-    let kind = if has_alloc_error_handler { AllocatorKind::Global } else { AllocatorKind::Default };
-    let callee = kind.fn_name(sym::oom);
+    let callee = if has_alloc_error_handler { "__rg_oom" } else { "__rdl_oom" };
```

Comment:
nit: it seems bad to embed the knowledge of what prefix to use directly into the code. Elsewhere, you have helper functions that add the appropriate prefix to use; why not apply that pattern here? Is it because it wouldn't get used elsewhere in the compiler?

Reply:
I don't quite recall. I will take a look tomorrow and maybe revert this.

Reply:
Added a helper function. The existing `AllocatorKind::fn_name` helper was changed to use `__rust_*` for AllocatorKind::Global, but this still needs `__rg_oom`.